### ISSUE TITLE
Torch in IOS

### ISSUE THREAD

**Issue by Diogo1995** [zxing-js/ngx-scanner#168]
Torch can't be turned On on Iphones. Is it possible to fix this? :)

**Comment by odahcam** [comment]
Can you share the iOS and Safari versions you used to test that?

**Comment by Diogo1995** [comment]
IOs 12.1.4
safari 12

**Comment by csutorasr** [comment]
Can you try the app in [this answer](https://stackoverflow.com/a/47153547/1558028)? Does it show anything for you?

**Comment by Diogo1995** [comment]
nop, not working..

________________________________
De: Robin Csutorás <notifications@github.com>
Enviado: segunda-feira, 25 de fevereiro de 2019 17:06
Para: zxing-js/ngx-scanner
Cc: DiogoPedrosa; Author
Assunto: Re: [zxing-js/ngx-scanner] Torch in IOS (#168)


Can you try the app in this answer<https://stackoverflow.com/a/47153547/1558028>? Does it show anything for you?

**Comment by odahcam** [comment]
Is there any update available for your device?

**Comment by Diogo1995** [comment]
no, there isn't. I have iOS 12.1.4

________________________________
De: Luiz Machado <notifications@github.com>
Enviado: segunda-feira, 25 de fevereiro de 2019 22:39
Para: zxing-js/ngx-scanner
Cc: DiogoPedrosa; Author
Assunto: Re: [zxing-js/ngx-scanner] Torch in IOS (#168)


Is there any update available for your device?

**Comment by martinpereiranm** [comment]
Is there any update on this?

**Comment by csutorasr** [comment]
@odahcam Can you get an iPhone to test it?
@Diogo1995  @martinpereiranm Which devices are these?

**Comment by Diogo1995** [comment]
@csutorasr IPhone 7

**Comment by martinpereiranm** [comment]
Me too

**Comment by Neknail** [comment]
Me too. Is this bug fixed already?

**Comment by odahcam** [comment]
Nope, not fixed yet. I'll try to get an iPhone this week to test it and see if the torch works, but, unfortunately I do not have an iPhone currently available for debugging/developing purposes.

**Comment by GuoDaFeng666** [comment]
I have same problem,Is it settled now？

**Comment by odahcam** [comment]
Hey, sorry for the delay (covid didn't got me yet). I have bad news everyone, both `torch` and `fillLightMode` **are not** available for browsers other than Chrome AFAIK. As every browser in iOS but Safari runs in WkWebView, it's highly improbable that this would work in this system at all.

Another point, I need to ask you to not rely on this torch feature as it is completely experimental and only manually tested in the hands of the developers who implemented it and I being one of them I can assure you this feature is not production ready.

Any info or help in this matter is welcome.

Also, I'm transferring this issue to the right place, as torch support is dependent in the @zxing-js/browser and is only used by this package.
